src/CoreWCF.Primitives/src/CoreWCF/Channels/WindowsStreamSecurityUpgradeProvider.cs

```diff
@@ -199,13 +199,13 @@ protected override async Task<(Stream, SecurityMessageProperty)> OnAcceptUpgrade
                 return clientSecurity;
             }
 
-            public override SecurityMessageProperty GetRemoteSecurity()
+            public override Task<SecurityMessageProperty> GetRemoteSecurityAsync()
             {
```

Comment:
By keeping GetRemoteSecurity in the base class and making GetRemoteSecurityAsync virtual and have it call GetRemoteSecurity and wrap the result in Task.FromResult, that means implementations which actually are synchronous like this one can just implement GetRemoteSecurity.

Reply:
I removed the GetRemoteSecurityAsync API I intoduced before as you investigated and noticed this API does not need to be async.